src/utils/svelte.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Indicates whether the attribute value is a Svelte interpolation.
+ *
+ * ---
+ *
+ * Example interpolation:
+ * ```
+ * a(href="{ cat.id }")
+ * ```
+ *
+ * In this case `val` is `"{ cat.id }"`.
+ *
+ * ---
+ *
+ * @param val Value of tag attribute.
+ * @returns `true` if `val` passes the svelte interpolation check, otherwise `false`.
+ */
+export function isSvelteInterpolation(val: string): boolean {
+	return (
+		val.length >= 3 &&
+		((val[0] === '"' && val[val.length - 1] === '"') || (val[0] === "'" && val[val.length - 1] === "'")) &&
```

Comment:
These expressions should be simplified into a separate method calls as well.

````js
return (
    val.length >= 3 &&
    (isWrappedWith(val, '"') || isWrappedWith(val, "'")) &&
    isWrappedWith(val, '{', '}', 1) &&
    !val.includes('{', 2)
);
````


````js
function isWrappedWith(val: string, wrapper: string, offset: int = 0) {
	return function isWrappedWith(val, wrapper, wrapper, offset);
}

function isWrappedWith(val: string, start: string, end: string, offset: int = 0) {
    return val[offset] === start && val[val.length - 1 - offset] === end;
}
````

Reply:
Maybe you can even add something like this:

````js
function isQuoted(val: string, offset: int = 1) {
    return isWrappedWith(val, '"', offset) || isWrappedWith(val, "'", offset);
}
````

I'm sure you needed something like this in many places.